tekton-catalog/pipeline-loops/pkg/reconciler/pipelinelooprun/pipelinelooprun.go

```diff
@@ -597,53 +601,64 @@ func computeIterations(run *v1alpha1.Run, tls *pipelineloopv1alpha1.PipelineLoop
 			if p.Name == "to" {
 				to, _ = strconv.Atoi(p.Value.StringVal)
 			}
-		} else if p.Name == tls.IterateParam {
-			if p.Value.Type == v1beta1.ParamTypeString {
-				// Transfer p.Value to Array.
-				var strings []string
-				var ints []int
-				var dictsString []map[string]string
-				var dictsInt []map[string]int
-				errString := json.Unmarshal([]byte(p.Value.StringVal), &strings)
-				errInt := json.Unmarshal([]byte(p.Value.StringVal), &ints)
-				errDictString := json.Unmarshal([]byte(p.Value.StringVal), &dictsString)
-				errDictInt := json.Unmarshal([]byte(p.Value.StringVal), &dictsInt)
-				if errString != nil && errInt != nil && errDictString != nil && errDictInt != nil {
-					return 0, iterationElements, fmt.Errorf("The value of the iterate parameter %q can not transfer to array", tls.IterateParam)
+		} else {
+			if p.Name == tls.IterateParam {
+				if p.Value.Type == v1beta1.ParamTypeString {
+					iterationParamStr = p.Value.StringVal
 				}
-
-				if errString == nil {
-					numberOfIterations = len(strings)
-					for _, v := range strings {
-						iterationElements = append(iterationElements, v)
-					}
-					break
-				} else if errInt == nil {
-					numberOfIterations = len(ints)
-					for _, v := range ints {
-						iterationElements = append(iterationElements, v)
-					}
-					break
-				} else if errDictString == nil {
-					numberOfIterations = len(dictsString)
-					for _, v := range dictsString {
-						iterationElements = append(iterationElements, v)
-					}
-					break
-				} else if errDictInt == nil {
-					numberOfIterations = len(dictsInt)
-					for _, v := range dictsInt {
+				if p.Value.Type == v1beta1.ParamTypeArray {
+					numberOfIterations = len(p.Value.ArrayVal)
+					for _, v := range p.Value.ArrayVal {
 						iterationElements = append(iterationElements, v)
 					}
 					break
 				}
 			}
-			if p.Value.Type == v1beta1.ParamTypeArray {
-				numberOfIterations = len(p.Value.ArrayVal)
-				for _, v := range p.Value.ArrayVal {
+			if p.Name == tls.IterateParamStringSeparator {
+				iterationParamStrSeparator = p.Value.StringVal
+			}
+		}
+	}
+	if iterationParamStr != "" {
```

Comment:
Is white space allowed? In yaml it should be " "?

Reply:
allowed, I added the unit test case